shared/domains/chelonia/chelonia.js

```diff
@@ -1377,30 +1377,48 @@ export default (sbp('sbp/selectors/register', {
   'chelonia/kv/set': async function (contractID: string, key: string, data: Object, {
     innerSigningKeyId,
     encryptionKeyId,
-    signingKeyId
+    signingKeyId,
+    maxAttempts = 3,
+    onconflict
   }: {
     innerSigningKeyId: ?string,
     encryptionKeyId: ?string,
-    signingKeyId: string
+    signingKeyId: string,
+    maxAttempts: ?number,
+    onconflict: (contractID: string, key: string, data: Object) => Promise<Object>,
   }) {
-    const serializedData = outputEncryptedOrUnencryptedMessage.call(this, {
-      contractID,
-      innerSigningKeyId,
-      encryptionKeyId,
-      signingKeyId,
-      data,
-      meta: key
-    })
-    const response = await fetch(`${this.config.connectionURL}/kv/${encodeURIComponent(contractID)}/${encodeURIComponent(key)}`, {
-      headers: new Headers([[
-        'authorization', buildShelterAuthorizationHeader.call(this, contractID)
-      ]]),
-      method: 'POST',
-      body: JSON.stringify(serializedData),
-      signal: this.abortController.signal
-    })
-    if (!response.ok) {
-      throw new Error('Invalid response status: ' + response.status)
+    for (;;) {
+      const serializedData = outputEncryptedOrUnencryptedMessage.call(this, {
+        contractID,
+        innerSigningKeyId,
+        encryptionKeyId,
+        signingKeyId,
+        data,
+        meta: key
+      })
+      const response = await fetch(`${this.config.connectionURL}/kv/${encodeURIComponent(contractID)}/${encodeURIComponent(key)}`, {
+        headers: new Headers([[
+          'authorization', buildShelterAuthorizationHeader.call(this, contractID)
+        ]]),
+        method: 'POST',
+        body: JSON.stringify(serializedData),
+        signal: this.abortController.signal
+      })
+      if (!response.ok) {
+        if (response.status === 409) {
+          if (--maxAttempts <= 0) {
+            throw new Error('Conflict setting KV value')
+          }
+          await new Promise(resolve => setTimeout(resolve, randomIntFromRange(0, 1500)))
```

Comment:
Since we're using `randomIntFromRange` here from `giLodash`, we might as well import `delay` instead to shorten this code (as it's exactly for this purpose)